internal/dataplane/kong_client.go

```diff
@@ -311,13 +315,18 @@ func (c *KongClient) Update(ctx context.Context) error {
 	}
 
 	// parse the Kubernetes objects from the storer into Kong configuration
-	kongstate := p.Build()
-	// todo: does it still make sense to report TranslationCount when Build no longer returns an error?
-	// https://github.com/Kong/kubernetes-ingress-controller/issues/1892
-	c.prometheusMetrics.TranslationCount.With(prometheus.Labels{
-		metrics.SuccessKey: metrics.SuccessTrue,
-	}).Inc()
-	c.logger.Debug("successfully built data-plane configuration")
+	kongstate, translationFailures := p.Build()
+	if len(translationFailures) > 0 {
+		c.prometheusMetrics.TranslationCount.With(prometheus.Labels{
+			metrics.SuccessKey: metrics.SuccessFalse,
+		}).Inc()
+		c.logger.Debugf("%d translation failures have occurred when building data-plane configuration")
+	} else {
+		c.prometheusMetrics.TranslationCount.With(prometheus.Labels{
+			metrics.SuccessKey: metrics.SuccessTrue,
+		}).Inc()
+		c.logger.Debug("successfully built data-plane configuration")
+	}
```

Comment:
The `translationFailures` is not going to be used or it's going to be done in next PR?

Reply:
It's going to be used in the next PR, yup.

Reply:
It'd be cool to have a plan for the PRs that are going to be implemented to fix the root issue. So that there is an overview of that. WDYT?

Reply:
Yeah, that's a very good idea. I've added a "Suggested split of code changes" section to the root issue: https://github.com/Kong/kubernetes-ingress-controller/issues/3097, is that what you would expect?

Reply:
yep, that's exactly what I meant. TY!